examples/force_control/Force_control_example.py

```diff
@@ -0,0 +1,236 @@
+        denominator = M + D*dt + K*np.square(dt)
+        detx_ = numerator / denominator
+        print (detx_)
+        detx[2] = detx[1]
+        detx[1] = detx[0]
+        detx[0] = detx_
+
+        zzz = sphere.position[2] + detx[0]
+
+        # (0.46,0) is the the centre of the circle trajectory on the table
+        sphere.position = np.array([0.46 - r * np.sin(w * t + np.pi / 2), r * np.cos(w * t + np.pi / 2), zzz])
+        dv = kp * (sphere.position - x) - kd * dx  # compute the other direction tracking error term
+
+
+        sp_z.append(sphere.position[2])
+        num.append(t)
+
+        dw = kp * quaternion_error(sphere.orientation, o) - kd * do
+        # evaluate damped-least-squares IK
+        dq = Jp.dot(np.hstack((dv, dw)))
+
+        # set joint positions
+        q = q[qIdx] + dq * dt
+        robot.set_joint_positions(q, joint_ids=joint_ids)
+
+        # print(Fz_error, dv[2])
+        if t == 800:
+            break
+        # step in simulation
+        world.step(sleep_dt=dt)
+    plt.plot(num, sp_z)
+    plt.show()
+
+
+
+if __name__=='__main__':
+    # Create simulator
+    sim = Bullet()
+
+    # create world
+    world = BasicWorld(sim)
+
+    # flag : 0 # PI control
+    flag = 0
+    # create robot
+    robot = KukaIIWA(sim)
+    robot.print_info()
+    world.load_robot(robot)
+    world.load_table(position=np.array([1, 0., 0.]), orientation=np.array([0.0, 0.0, 0.0, 1.0]))
+    # define useful variables for IK
+    dt = 1. / 240
+    link_id = robot.get_end_effector_ids(end_effector=0)
+    joint_ids = robot.joints  # actuated joint
+    damping = 0.01  # for damped-least-squares IK
+    wrt_link_id = -1  # robot.get_link_ids('iiwa_link_1')
+    qIdx = robot.get_q_indices(joint_ids)
+
+    # define gains
+    kp = 500  # 5 if velocity control, 50 if position control
+    kd = 5  # 2*np.sqrt(kp)
+
+    # create sphere to follow
+    sphere = world.load_visual_sphere(position=np.array([0.5, 0., 0.5]), radius=0.05, color=(1, 0, 0, 0.5))
+    sphere = Body(sim, body_id=sphere)
+
+    # set initial joint p
+    # ositions (based on the position of the sphere at [0.5, 0, 1])
+    robot.reset_joint_states(q=[8.84305270e-05, 7.11378917e-02, -1.68059886e-04, -9.71690439e-01, 1.68308810e-05,
+                                3.71467111e-01, 5.62890805e-05])
+
+    # define amplitude and angular velocity when moving the sphere
+    w = 0.01
+    r = 0.1
+
+    # I set the reference orientation to a constant
+    sphere.orientation = np.array([1, 0, 0, 0])
+
+    FT_sensor = sensors.JointForceTorqueSensor(sim, body_id=robot.id, joint_ids=6)
+    # The plotting handle
+    plot = EeFtRealTimePlot(robot, sensor=FT_sensor, forcex=True, forcey=True, forcez=True,
+                            torquex=True, torquey=True, torquez=True, num_point=1000, ticks=24)
+    # FT_ = np.zeros(6)
+
+    plot_t = Thread(target=plotting_thread, args=[plot], name='plotting task')
```

Comment:
May you please explain the need to have threads here? I mean, I am curious about the motivation of using threads here for plotting and controlling the robot.